tests/functional/dataplane/openstackdataplanenodeset_controller_test.go

```diff
@@ -1435,6 +1436,72 @@ var _ = Describe("Dataplane NodeSet Test", func() {
 		})
 	})
 
+	When("A NodeSet and Deployment are created with extraMounts with pvc template", func() {
+		BeforeEach(func() {
+			edpmVolClaimTemplate := corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: "edpm-ansible",
+				ReadOnly:  true,
+			}
+			nodeSetSpec := DefaultDataPlaneNodeSetSpec("edpm-compute")
+			nodeSetSpec["preProvisioned"] = true
+			nodeSetSpec["services"] = []string{"bootstrap"}
+			nodeSetSpec["nodeTemplate"] = map[string]interface{}{
+				"extraMounts": []storage.VolMounts{
+					{
+						Mounts: []corev1.VolumeMount{
+							{
+								Name:      "edpm-ansible",
+								MountPath: "/usr/share/ansible/collections/ansible_collections/osp/edpm",
+							},
+						},
+						Volumes: []corev1.Volume{
```

Comment:
After https://github.com/openstack-k8s-operators/openstack-operator/pull/1158 merges, I think this structure here requires a change and it should be:

```golang
Volumes: []storage.Volume {
{
    Name: "edpm-ansible",
    VolumeSource: storage.VolumeSource {
        PersistentVolumeClaim: &edpmVolClaimTemplate,
    },						
},
...
```

I'm not sure you want to do the other way round, like merge this patch and update the existing #1158 accordingly.

Reply:
Yeah, merge the other patch first and I'll rebase this one.

I just created this because debugging the issue in Kuttl was difficult while debugging in functional tests is generally easier for me. I'll change this to draft while we wait for the other one.